src/vcf/fixer.cpp

```diff
@@ -128,10 +162,10 @@ namespace ebi
             };
 
             fix_column(info_column_index, string_line, "\t", [&](std::string &info_column) {
-                num_removed_fields = remove_column(info_column, ";", empty_info_column, condition_to_remove_info_field);
+                num_removed_fields = remove_column(info_column, ";", empty_info_column, expected_info_column, condition_to_remove_info_field);
```

Comment:
same as above, split the paths, one for removing a INFO field, another path for replacing a value maybe another for duplicated fields. right now is hard to tell how many fixes this code is performing.

 I'm starting to wonder if this should be expressed explicitly as an enum or something, like `enum {duplication, irrecoverable_value, recoverable_value }`. before moving into that, let's see how this results with several `if`s.